File: app/src/main/java/example/develop/davidoh/java_android_mvp_example/data/source/flickr/FlickrRepository.java
package example.develop.davidoh.java_android_mvp_example.data.source.flickr;

public class FlickrRepository implements FlickrDataSource{
    private static FlickrRepository flickerRepository;

    private FlickrRemoteData flickrRemoteData;

    public static FlickrRepository getInstance(){
        if (flickerRepository == null){
            synchronized (FlickrRepository.class){
                if (flickerRepository == null){
                    flickerRepository = new FlickrRepository();
                }
            }
        }
        return flickerRepository;
    }

    private FlickrRepository() {
        flickrRemoteData = FlickrRemoteData.getInstance();
    }


    @Override
    public void getRecentPhoto(int page, int perPage, LoadImageCallback loadImageCallback) {
        flickrRemoteData.getRecentPhoto(page, perPage, loadImageCallback);
    }

    @Override
    public void getSearchPhoto(String keyword, int page, int perPage, LoadImageCallback loadImageCallback) {
        flickrRemoteData.getSearchPhoto(keyword, page, perPage, loadImageCallback);
    }
}

File: app/src/main/java/example/develop/davidoh/java_android_mvp_example/data/source/flickr/FlickrRepository.kt
package example.develop.davidoh.java_android_mvp_example.data.source.flickr

import example.develop.davidoh.java_android_mvp_example.data.PhotoResponse
import retrofit2.Call

object FlickrRepository : FlickrDataSource {

    private val flickrRemoteData = FlickrRemoteData()

    override fun getRecentPhoto(page: Int, perPage: Int): Call<PhotoResponse> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getSearchPhoto(keyword: String, page: Int, perPage: Int): Call<PhotoResponse>
            = flickrRemoteData.getSearchPhoto(keyword, page, perPage)

}
